tests/app/questionnaire/test_placeholder_parser.py

```diff
@@ -27,7 +31,16 @@ def test_metadata_placeholder(mock_renderer):
         language="en",
         answer_store=AnswerStore(),
         list_store=ListStore(),
-        metadata={"period_str": period_str},
+        metadata=MetadataProxy.from_dict(
+            {
+                "period_str": period_str,
+                "tx_id": "tx_id",
+                "account_service_url": "account_service_url",
+                "response_id": "response_id",
+                "collection_exercise_sid": "collection_exercise_sid",
+                "case_id": "case_id",
+            }
+        ),
```

Comment:
I'm leaving this here but applies to a lot of test file. There is a lot of duplication for metadata. I suggest creating a method for this. So given a method:
```py
def get_metadata(extra_metadata: dict = None):
    extra_metadata = extra_metadata or {}
    metadata = {
        "response_id": "1",
        "account_service_url": "account_service_url",
        "tx_id": "tx_id",
        "collection_exercise_sid": "collection_exercise_sid",
        "case_id": "case_id",
        **extra_metadata
    }
    return MetadataProxy.from_dict(metadata)
```
This allows you to even override mandatory metadata. With this, this would become:
```py
metadata=get_metadata({"period_str": period_str}),
```

Reply:
Updated 👍